### PR TITLE
feat(NES-1601): default Copy To Team dialog to active team

### PR DESCRIPTION
**Pull request by jianwei1**
## Summary

- Pre-fill the team selector in the *Add Journey to Team* dialog with the user's **active team** when they click **Use This Template** on a team template — addresses the extra click Lucinda flagged ([NES-1601](https://linear.app/jesus-film-project/issue/NES-1601/set-dialog-default-to-current-team-when-using-team-templates), Slack [thread](https://jfp-digital.slack.com/archives/C08GT3TGLDP/p1776956602585379)).
- Behaviour change is **opt-in**, scoped to the team-template *Use This Template* flow only. The shared `CopyToTeamDialog` is also used by *Copy to ...* and *Duplicate*, both of which keep their existing empty-default behaviour.
- Fully client-side. The `JourneyDuplicate` mutation already accepts `teamId` from the client, so **no `api-journeys` changes** are needed.

## What changed

- **`libs/journeys/ui/src/components/CopyToTeamDialog/CopyToTeamDialog.tsx`** — added an opt-in `defaultToActiveTeam?: boolean` prop. When `true`, `initialValues.teamSelect` resolves to `useTeam().activeTeam?.id` before the existing `teams.length === 1 ? teams[0].id : ''` fallback. When `false` (the default) or omitted, behaviour is unchanged. `enableReinitialize` is already on, so Formik handles the async `activeTeam` resolution cleanly.
- **`libs/journeys/ui/src/components/TemplateView/CreateJourneyButton/CreateJourneyButton.tsx`** — extends `JourneyForTemplate` with `team?: { id: string } | null` and passes `defaultToActiveTeam={journeyDataToUse?.team != null && journeyDataToUse.team.id !== 'jfp-team'}`. Discriminator mirrors `DefaultMenu.isLocalTemplate` so the team-vs-global-JFP check is consistent across the app.
- **`CopyToTeamMenuItem.tsx`** and **`DuplicateJourneyMenuItem.tsx`** — untouched. Their flows keep the empty default (which is the right UX for cross-team copy / shared-view duplicate).

### Why narrow rather than broad

The dialog has three production callers. *Copy to ...* is semantically a cross-team copy where the empty default forces a deliberate pick — pre-filling there would mask intent. *Duplicate* only opens the dialog when `activeTeam === null`, where the new default would be a no-op anyway. So the opt-in prop targets the only flow where Lucinda's complaint applies.

## Testing

- **`CopyToTeamDialog.spec.tsx`** — 3 new tests in a `defaultToActiveTeam prop` block: flag-on + active team → pre-fill, flag-on + null active team → empty, flag-off + active team → empty (proves opt-in semantics). 30 existing tests pass unchanged. **Total: 33.**
- **`CreateJourneyButton.spec.tsx`** — 3 new tests in a `defaultToActiveTeam wiring` block: team template (`team.id !== 'jfp-team'`) → pre-fill, global JFP (`team.id === 'jfp-team'`) → empty, `team: null` → empty. 36 existing tests pass unchanged. **Total: 39.**
- `CopyToTeamMenuItem.spec.tsx` (9), `DuplicateJourneyMenuItem.spec.tsx` (6), and the broader `TemplateView` suite (19 files, 159 tests) all pass — confirming the other dialog callers are unaffected.
- `nx lint journeys-ui` clean (no new warnings). `nx type-check journeys-ui` clean.

## Plan

[`docs/plans/2026-04-24-001-feat-default-copy-to-team-dialog-to-active-team-plan.md`](https://github.com/JesusFilm/core/blob/jianweichong/nes-1601-set-dialog-default-to-current-team-when-using-team-templates/docs/plans/2026-04-24-001-feat-default-copy-to-team-dialog-to-active-team-plan.md) tracks the design rationale, alternatives considered, and acceptance criteria.

## Post-Deploy Monitoring & Validation

No additional operational monitoring required: this is a client-only default-value change on a form field, opt-in via a new prop and gated to a single caller. The mutation shape and payloads are unchanged, there are no new error paths, and the fallback preserves prior behaviour when `activeTeam` is not resolved.

## Release Notes

* **New Features**
  * When copying team-specific templates, the team selection dialog now defaults to your active team, streamlining the workflow with fewer required steps.

* **Tests**
  * Added comprehensive test coverage for team selection defaulting behavior and template routing scenarios.

### CODE REVIEW

**Comment by csiyang** [comment]
Reviewed and tested locally — implementation is correct and the scope is properly narrowed per Lucinda's ask. Approving from my end.

### What I verified

**Scope check** — three production callers of `CopyToTeamDialog`:
- `CreateJourneyButton` — modified, opt-in correctly gated by `journeyDataToUse?.team != null && journeyDataToUse.team.id !== 'jfp-team'`. Discriminator mirrors the existing `isLocalTemplate` definition in `DefaultMenu.tsx`, so it's consistent with how the rest of the app distinguishes team vs global JFP templates.
- `CopyToTeamMenuItem` — unchanged. Doesn't pass the new prop → default `false` → behaviour preserved. Verified `CopyToTeamMenuItem.spec.tsx` (9 tests) still passes.
- `DuplicateJourneyMenuItem` — unchanged. Same as above. `DuplicateJourneyMenuItem.spec.tsx` (6 tests) still passes.

**Edge cases** (all handled by the `?? fallback` chain):
- `activeTeam` undefined (loading) → falls through to existing default.
- `activeTeam` null (Shared with me view) → falls through.
- Single-team user → still pre-fills with that team (existing behaviour).
- Template owned by team-c viewed by team-a user → defaults to team-a (matches Lucinda's "the team you are on" wording).

**Tests** — both positive and negative cases covered:
- Dialog: flag-on + active team → pre-fill, flag-on + null active team → empty, flag-off + active team → empty (proves opt-in semantics).
- Wiring: team template → pre-fill, `jfp-team` → no pre-fill, `team: null` → no pre-fill.

`nx type-check journeys-ui` clean. All four affected spec files green locally.

### One nit — PR description is stale

The body still describes the original broad-scope approach:
> `initialValues.teamSelect` now defaults to \`activeTeam?.id ?? (teams.length === 1 ? teams[0].id : '')\`.
> ...
> \### Scope note
> ... All usage sites share the better default.

The actual code introduces the `defaultToActiveTeam?: boolean` opt-in prop and gates it on the team-template flow only. Worth updating the "What changed" and "Scope note" sections so future readers (and anyone landing here from Linear) don't get whiplash between the description and the diff. The plan doc in `docs/plans/...` is up to date — the PR body just needs to catch up.

Not blocking — happy to merge once you've had a chance to refresh the description.

**Comment by jianwei1** [comment]
## Review feedback addressed (5f66e6eee)

**Fixed:**
- PR description — refreshed to describe the narrowed `defaultToActiveTeam` opt-in approach. The "What changed" section now reflects the prop, the wiring in `CreateJourneyButton`, and the unchanged callers; the "Scope note" describing the broad approach is replaced with "Why narrow rather than broad". Test counts updated (33 dialog, 39 CreateJourneyButton). Plan doc was already current — only the PR body needed to catch up.

Thanks for the careful review @csiyang 🙏

File: docs/plans/2026-04-24-001-feat-default-copy-to-team-dialog-to-active-team-plan.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+- No SSR risk: dialog is client-only (modal).
+
+## System-Wide Impact
+
+- **Interaction graph:** User clicks _Use this Template_ (or _Duplicate_ in a journey menu) → dialog opens → Formik initializes with new default → user clicks Submit → `submitAction` → `JourneyDuplicate` mutation → `updateLastActiveTeamId` mutation → `GetAdminJourneys` refetch. The chain is unchanged; only the initial field value changes.
+- **Error propagation:** No new error paths. Form validation on `teamSelect` (`required`, line 156) still fires if somehow the default resolves to `''`.
+- **State lifecycle risks:** None. `resetForm()` is still called on submit (line 140) and on close (line 210), so subsequent opens pick up the latest `activeTeam` cleanly.
+- **API surface parity:** No mutation signature changes. Both callers (`CopyToTeamMenuItem`, `DuplicateJourneyMenuItem`) inherit the new default without code changes.
+- **Integration test scenarios:** See **Acceptance Criteria** below; the new "multi-team user opens dialog" scenario is the one unit tests must cover.
```
[the remark this answers is not in the document]
Comment (jianwei1) [reply docs/plans/2026-04-24-001-feat-default-copy-to-team-dialog-to-active-team-plan.md:69]: @coderabbitai both points are already reflected in the current revision — line 68 already states the prop is opt-in and explicitly names `CopyToTeamMenuItem`/`DuplicateJourneyMenuItem` as unaffected, and the Acceptance Criteria covers flag-on/flag-off behavior in `CopyToTeamDialog.spec.tsx` (item 3 is the flag-off case). Resolving.

**Review by edmonday — APPROVED** [review]
